src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -38,7 +38,10 @@ def __init__(self, env_layer, composite_logger, events_folder_path):
         self.start_time_for_event_count_throttle_check = datetime.datetime.utcnow()
         self.event_count = 1
 
-        if events_folder_path is not None and os.path.exists(events_folder_path):
+        if events_folder_path is not None:
+            if not os.path.exists(events_folder_path):
+                os.mkdir(events_folder_path) # ensure events folder exists, even though telemetry should have created it
```

Comment:
Please do this in extension handler (/src/ext/src) when the value is first read.

Core code remains the same, and should fail as it does today if the handler did not help tide over the agent issue.

Reply:
Add quick UT in ext handler to ensure creation if missing.

Reply:
Thanks, moved to TelemetryWriter in extension.